src/cordova/apphost.js

```diff
@@ -234,19 +234,22 @@ define(['appStorage', 'browser'], function (appStorage, browser) {
         init: function() {
             return new Promise(function(resolve) {
                 document.addEventListener("deviceready", function () {
-                    // remove special characters
-                    deviceName = MainActivity.getDeviceModel().replace(/[^\w\s]/gi, '');
-                    deviceId = MainActivity.getDeviceId();
-                    appName = "Jellyfin Mobile";
-                    appVersion = MainActivity.getAppVersion();
-
-                    appInfo = {
-                        deviceId: deviceId,
-                        deviceName: deviceName,
-                        appVersion: appVersion
-                    };
-
-                    resolve(appInfo);
+                    window.NativeShell.getDeviceInformation(function(result) {
+                        deviceId = result.deviceId;
+                        deviceName = result.deviceName;
+                        appName = result.appName;
+                        appVersion = result.appVersion;
+                        appInfo = {
```

Comment:
why make separate variables instead of directly referencing them in `appInfo` like
```js
appInfo = {
    deviceId: result.deviceId,
    deviceName: result.deviceName,
// etc
}
```
?

Reply:
I tried that first but those are global variables that are used in other methods within the class, so I need to set them before returning them in the promise.

Reply:
Add this as a comment please, otherwise it's pretty unclear what's going on there